test/Microsoft.Extensions.Logging.EventSource.Test/EventSourceLoggerTest.cs

```diff
@@ -0,0 +1,581 @@
+
+                logger2.LogCritical(new EventId(5), new Exception("oops", new Exception("inner oops")),
+                    "Logger2 Event5 Critical {stringParam} {int1Param} {int2Param}", "bar", 23, 45);
+
+                using (logger3.BeginScope("Inner scope {timeParam} {guidParam}", TimeParam, GuidParam))
+                {
+                    logger2.LogWarning(new EventId(6), "Logger2 Event6 Warning NoParams");
+                }
+
+                logger3.LogInformation(new EventId(7), "Logger3 Event7 Information {stringParam} {doubleParam} {intParam}", "inner scope closed", DoubleParam2, 37);
+            }
+
+            logger2.LogWarning(new EventId(8), "Logger2 Event8 Warning {stringParam} {timeParam}", "Outer scope closed", TimeParam.ToString("O"));
+        }
+
+        private static void VerifyEvents(TestEventListener eventListener, params string[] verifierIDs)
+        {
+            Assert.Collection(eventListener.Events, verifierIDs.Select(id => EventVerifiers[id]).ToArray());
+        }
+
+        private static void VerifySingleEvent(string eventJson, string loggerName, string eventName, int? eventId, LogLevel? level, params string[] fragments)
+        {
+            Assert.True(eventJson.Contains(@"""__EVENT_NAME"":""" + eventName + @""""), $"Event name does not match. Expected {eventName}, event data is '{eventJson}'");
+            Assert.True(eventJson.Contains(@"""LoggerName"":""" + loggerName + @""""), $"Logger name does not match. Expected {loggerName}, event data is '{eventJson}'");
+            if (level.HasValue)
+            {
+                Assert.True(eventJson.Contains(@"""Level"":" + ((int)level.Value).ToString()), $"Log level does not match. Expected level {((int)level.Value).ToString()}, event data is '{eventJson}'");
+            }
+            if (eventId.HasValue)
+            {
+                Assert.True(eventJson.Contains(@"""EventId"":""" + eventId.Value.ToString()), $"Event id does not match. Expected id {eventId.Value}, event data is '{eventJson}'");
+            }
+
+            for (int i=0; i<fragments.Length; i++)
+            {
+                Assert.True(eventJson.Contains(fragments[i]), $"Event data '{eventJson}' does not contain expected fragment {fragments[i]}");
+            }
+        }
+
+        private class TestEventListener : EventListener
+        {
+            // This is a workaround for a bug in EventListener where OnEventSourceCreated will be called before the listener
+            // is fully constructed. Unfortunately this bug has shipped and is now an "expected behavior".
+            public class ListenerSettings
+            {
+                public EventKeywords Keywords;
+                public EventLevel Level;
+                public string FilterSpec;
+            }
+            public static ListenerSettings Settings = new ListenerSettings();
```

Comment:
Let me make sure that we are on the same page.

The fact that an EventSource with a given name is a singleton is not a developer's choice, it is part of EventSource contract. For example

[EventSource(Name="MyCompany-ApplicationX-MyEventSource")]
public sealed class MyEventSource: EventSource { ... }

var es1 = new MyEventSource();
var es2 = new MyEventSource(); // This line will _always_ throw an exception--there can be only one MyEventSource in the process

I guess if we really need to make the tests run in parallel, we could do that by adding an extra tag to every trace indicating to which test the given trace belongs. Then at EventListener level we could filter out all traces that do not belong to given test. The problem with this approach is that it makes tests more likely to miss a case when the bridge passes too many thing through. I'd rather keep things as straightforward as possible. 

I have not measured it, but it seems that the speed of execution of this test suite is below discernable threshold. How much do we care about saving, say, 300 ms?